src/worker/ingestion/team-manager.ts

```diff
@@ -101,42 +100,35 @@ export class TeamManager {
             event: event,
             ingested: team.ingested_event,
         })
-        let shouldUpdate = this.calculateUpdates(team, event, properties)
-        if (shouldUpdate && team.__fetch_event_uuid !== eventUuid) {
-            // :TRICKY: Double-check if we're updating based on cached data, if so, re-validate.
-            // :TODO: Switch all of this to a sane schema that can be updated without races.
-            this.teamCache.delete(teamId)
-            team = await this.fetchTeam(teamId, eventUuid)
-            shouldUpdate = this.calculateUpdates(team, event, properties)
-        }
-        if (team && shouldUpdate) {
-            const timeout2 = timeoutGuard(
-                'Still running "updateEventNamesAndProperties" save. Timeout warning after 30 sec!',
-                { event }
-            )
+
+        await this.cacheEventNamesAndProperties(team.id)
+
+        if (!this.eventNamesCache.get(team.id)?.has(event)) {
             await this.db.postgresQuery(
-                `UPDATE posthog_team SET
-                    ingested_event = $1,
-                    event_names = $2,
-                    event_names_with_usage = $3,
-                    event_properties = $4,
-                    event_properties_with_usage = $5,
-                    event_properties_numerical = $6
-                WHERE id = $7`,
-                [
-                    true,
-                    JSON.stringify(team.event_names),
-                    JSON.stringify(team.event_names_with_usage),
-                    JSON.stringify(team.event_properties),
-                    JSON.stringify(team.event_properties_with_usage),
-                    JSON.stringify(team.event_properties_numerical),
-                    team.id,
-                ],
-                'updateEventNamesAndProperties'
+                `INSERT INTO posthog_eventdefinition (id, name, volume_30_day, query_usage_30_day, team_id) VALUES ($1, $2, NULL, NULL, $3) ON CONFLICT DO NOTHING`,
+                [new UUIDT().toString(), event, team.id],
+                'insertEventDefinition'
             )
-            clearTimeout(timeout2)
+            this.eventNamesCache.get(team.id)?.add(event)
+        }
+
+        for (const [key, value] of Object.entries(properties)) {
+            if (!this.eventPropertiesCache.get(team.id)?.has(key)) {
+                await this.db.postgresQuery(
+                    `INSERT INTO posthog_propertydefinition (id, name, is_numerical, volume_30_day, query_usage_30_day, team_id) VALUES ($1, $2, $3, NULL, NULL, $4) ON CONFLICT DO NOTHING`,
+                    [new UUIDT().toString(), key, typeof value === 'number', team.id],
+                    'insertPropertyDefinition'
+                )
+            }
         }
+
         if (team && !team.ingested_event) {
+            await this.db.postgresQuery(
+                `UPDATE posthog_team SET ingested_event = $1 WHERE id = $2`,
+                [true, team.id],
```

Comment:
comment just purely for my own learning here: is it just our own standard to have all values be passed as an array? Cause in this case `true` is not dynamic so it could be in the query directly. Not a complaint, just wondering for my own purposes - I do get the readability/standardization this brings

Reply:
No standard, just whatever does the job :)
In this specific case, I'm not sure why it's one way and not the other... and I think it doesn't really make any difference.